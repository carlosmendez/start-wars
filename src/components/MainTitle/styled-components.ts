import styled from '@emotion/styled';

export const StyledMainTitle = styled.div`
  text-align: center;
  h2 {
    color: rgb(255, 228, 0);
  }
`;
