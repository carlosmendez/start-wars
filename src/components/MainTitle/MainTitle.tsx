import React, { ReactNode } from 'react';
import { StyledMainTitle } from "./styled-components";

type MainTitleProps = {
  children: ReactNode;
};

const MainTitle = ({
  children,
}: MainTitleProps) => {
  return (
    <StyledMainTitle>
      <h1>STAR WARS</h1>
      {children}
    </StyledMainTitle>
  )
}

export { MainTitle };
