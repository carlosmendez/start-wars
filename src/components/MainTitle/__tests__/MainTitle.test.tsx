import React from 'react';
import renderer from 'react-test-renderer';
import { MainTitle } from "../MainTitle";

describe('MainTitle component', () => {
  it('renders correctly', () => {
    const jsx = (
      <MainTitle>mockTitle</MainTitle>
    );
    const tree = renderer
      .create(jsx)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
