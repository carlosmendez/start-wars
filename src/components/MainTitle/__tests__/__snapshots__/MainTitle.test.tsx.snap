// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTitle component renders correctly 1`] = `
<div
  className="css-1mxmwia"
>
  <h1>
    START WARS
  </h1>
  mockTitle
</div>
`;
